L46-core-tls-credential-API.md

```diff
@@ -0,0 +1,425 @@
+  // are required, and if the client certificate must be trusted.  The
+  // default is GRPC_SSL_DONT_REQUEST_CLIENT_CERTIFICATE.
+  void set_cert_request_type(
+      grpc_ssl_client_certificate_request_type cert_request_type);
+}
+
+// Client-specific options for configuring TLS.
+//
+// A client may optionally set a certificate provider.
+// If there is no certificate provider, the system default root certificates will be used to
+// verify server certificates. 
+// If a certificate provider is set and it provides root certifices that root
+// will be used.
+// If a certificate provider is set and it provides identity credentials those
+// identity credentials will be used.
+class TlsChannelCredentialsOptions final : public TlsCredentialsOptions {
+ public:
+  // Sets the decision of whether to do a crypto check on the server certificates.
+  // The default is true.
+  void set_verify_server_certificates(bool verify_server_certs);
+}
+
+// Builds a ChannelCredentials instance that establishes TLS connections in the manner specified by options.
+std::shared_ptr<ChannelCredentials> TlsCredentials(
+    const TlsChannelCredentialsOptions& options);
+}
+
+// Builds a ServerCredentials instance that establishes TLS connections in the manner specified by options.
+std::shared_ptr<ServerCredentials> TlsServerCredentials(
+    const TlsServerCredentialsOptions& options);
+```
+
+Here is an example of how to configure channel credentials:
+```c++
+/* Create option and set basic security primitives. */
+TlsChannelCredentialsOptions options = TlsChannelCredentialsOptions();
+options.set_verify_server_cert(true);
+options.set_min_tls_version(TLS1_2);
+options.set_max_tls_version(TLS1_3);
+
+/* Create TLS channel credentials. */
+std::shared_ptr<ChannelCredentials> creds = TlsCredentials(options);
+```
+
+### Credential Reloading
+Credential reloading will be implemented via a `CertificateProviderInterface`.
+This provider is responsible for sourcing key material and supplying it to the
+internal stack via the `CertificateProvider`'s `SetKeyMaterials` API.  Two
+reference implementations of the `CertificateProviderInterface` are provided -
+`StaticDataCertificateProvider` and `FileWatcherCertificateProvider`. The
+`StaticDataCertificateProvider` provides certificates from raw string data. The
+`FileWatcherCertificateProvider` will periodically and asynchronously refresh
+certificate data from a file, allowing changes to certificates without
+restarting the client/server. More detail about these reference implementations
+is provided below. These implementations should cover many use cases.  If these
+providers do not support the intricacies for a specific use case, a user can
+provide their own implementations of the `CertificateProviderInterface`.
+
+```c
+/* Opaque types. */
+// A struct that stores the credential data presented to the peer in handshake
+// to show local identity. The private_key and certificate_chain must be PEM
```

Comment:
micro-nit: We are no longer using these variable names, so let's just change them to "private key" and "certificate chain" without underscores?